### ISSUE TITLE
Rule-Based Styling of Line Layer Objects

### ISSUE THREAD

**Issue by AlexeyNAA** [qgis/QGIS#49748]
### What is the bug or the crash?

Rule-Based Styling of Line Layer Objects. Сreate a rule for the design of a linear layer object. Then we copy the rule several times in a row and immediately set the changes we need to the newly copied rules at the same time. In this case, the display of colors in the copied rules is incorrect.
If we copy the rule only once and immediately make the change we need, then no error occurs. linear objects of the layer with different colors based on the rules. 

### Steps to reproduce the issue

1. double click on line layer
2. click control feature symbology
3. click rule-based
![1](https://user-images.githubusercontent.com/110985567/184131684-74e56dac-94df-45c2-9a31-457d8d19c6a5.png)
4. creat any rule
5. click apply and ok
6. copy-paste the newly created rule 3-4 times in a row
7. make changes to the newly copied rules
8. we check the operation of the rules by disabling and enabling them

### Versions

QGIS version
3.22.9-Białowieża
QGIS code revision
a8e9e6fa
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.5.1
PROJ version
9.0.1
EPSG Registry database version
v10.064 (2022-05-19)
GEOS version
3.10.3-CAPI-1.16.1
SQLite version
3.38.1
PDAL version
2.3.0
PostgreSQL client version
14.3
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
CalculateGeometry
0.6.5
FreehandRasterGeoreferencer
0.8.3
HCMGIS
21.8.28
kmltools
3.1.21
latlontools
3.6.4
maplibrary
1.2
mapswipetool_plugin
1.2
mask
1.10.2
nominatim
1.4.2
numericalDigitize
0.4.7
osminfo
0.7.1
qgis-maptiler-plugin
2.0.0
QuickOSM
2.0.1
quick_map_services
0.19.29
SRTM-Downloader
3.1.17
db_manager
0.1.20
grassprovider
2.12.99
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by roya0045** [comment]
I'm not fully sure what is the issue, but my guess would be that this is a duplicate of #47915
